lib/absinthe/type/extensions.ex

```diff
@@ -0,0 +1,112 @@
+defmodule Absinthe.Type.Extensions do
+  use Absinthe.Schema.Notation
+
+  @moduledoc """
+  This module contains additional data types.
+
+  To use: `import_types Absinthe.Type.Extensions`.
+  """
+
+  scalar :datetime, name: "DateTime" do
+    description """
+    The `DateTime` scalar type represents a date and time in the UTC
+    timezone. The DateTime appears in a JSON response as an ISO8601 formatted
+    string, including UTC timezone ("Z").
+    """
+
+    serialize &DateTime.to_iso8601/1
+    parse parse_with([Absinthe.Blueprint.Input.DateTime], &parse_datetime/1)
```

Comment:
I think you're misunderstanding what `parse_with` does. These are all gonna be `Input.String` types because that's the Blueprint type that these datetime strings will initially come in as. They get parsed into datetimes sure, but prior to that they're just strings. The list you pass to `parse_with` is a list of the basic input types that are acceptable.